Rakefile

```diff
@@ -36,3 +36,36 @@ if File.exist?(build_defs_file)
     end
   end
 end
+
+desc "verify that commit messages match CONTRIBUTING.md requirements"
+task(:commits) do
+  commits = ENV['TRAVIS_COMMIT_RANGE'] || 'master..HEAD'
```

Comment:
So, rather than doing hacky things to try to get this working with shallow cloning, let's just use the fact that travis knows what commits are in the PR when doing to check.

Default to master..HEAD when running locally, which probably isn't quite what you want to do, but it should work.